### PR TITLE
For SPLAT-103: Add "Created Note" and "Created Reply" metrics

### PR DESCRIPTION
**Pull request by nmontmarquette**
Basically add base code for handling the "Created Note" and "Created Reply" metrics.
Further changes might populate some of the currently non-populated metric fields.

### CODE REVIEW

File: python/activity_stream/activity_stream.py [hunk 1]
```diff
@@ -14,6 +14,8 @@
 
 from sgtk.platform.qt import QtCore, QtGui
 
+from . import utils as qtwidgets_utils
```
Comment (josh-t) [inline python/activity_stream/activity_stream.py]: is this needed?

File: python/activity_stream/metric_utils.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2017 Shotgun Software Inc.
```
Comment (josh-t) [inline python/activity_stream/metric_utils.py]: So in the other PRs, the property gathering and metric logging was inline with the rest of the code. Here you've moved the actual logging to a separate util file. Suggest we just do this inline to be consistent unless there's a strong reason not to.

File: python/activity_stream/reply_list.py [hunk 3]
```diff
@@ -418,6 +420,7 @@ def _on_reply_clicked(self, note_id):
             self.__small_overlay.show()
             if reply_dialog.exec_() == QtGui.QDialog.Accepted:
                 self._data_manager.rescan()
+                log_metric_created_reply(self._bundle, "Reply List")
```
Comment (josh-t) [inline python/activity_stream/reply_list.py]: Should this be done in the `ReplyDialog`?

File: python/activity_stream/metric_utils.py (hunk 2 above)
Comment (nmontmarquette) [reply python/activity_stream/metric_utils.py]: The reason was that I needed to emit the same metric from different modules (2 for now) maybe more is the future?

File: python/activity_stream/reply_list.py (hunk 3 above)
Comment (nmontmarquette) [reply python/activity_stream/reply_list.py]: Would that be another one then? There is a field for source. One would be "Reply List" the one your are mentionning would be "Reply Dialog" ?

Comment (nmontmarquette) [reply python/activity_stream/reply_list.py]: How do I know I'm in "Reply List" from within the QT Dialog class ?

**Pull request merged**